### DISCUSSION TITLE
Post to /api/question

### DISCUSSION THREAD

**Discussion by gdziuba** [pashpashpash/vault-ai#90] [Q&A] [answered]
Can someone give me an example curl statement that will allow me ask questions?  Here is my curl statement:

`curl -X POST -H "Content-Type: application/json" -d '{"Question":"Who was the first president of the USA?"}' http://<<IP>>:8100/api/question`

I have been able to upload a file remotely, but haven't been able to ask it a question after uploading a file.

Thanks in advance.

**Comment by gdziuba** [comment, marked as answer]
I figured it out.  It was two parts.  I couldn't send the question in the form of json and I had the wrong endpoint.  Needed an s at the end.

/api/question
`curl -X POST -H "Content-Type: multipart/form-data" -F "Question=Who is the oldest little rascal?" -F "Model=GPT3Dot5Turbo" -F "UUID=<<My UUID>>" -F "ApiKey=<<MYAPI>>" http://<<IP>>:8100/api/questions`

and for those that are curious
/upload
`curl -X POST -H "Content-Type: multipart/form-data" -F "files=@/tmp/file.txt" -F "uuid=<<UUID>>" -F "apikey=<<API>>" http://<<API>>:5000/upload`

**Reply by BenutzerEinsZweiDrei** [reply]
Here is the api request for python

```python
import requests
import json
from requests_toolbelt import MultipartEncoder

def send():
	
	url = "https://vault.pash.city/api/questions"
	fields = { "Question": "What is a good answer?", "Model": "GPT3Dot5Turbo", "UUID" : "7364aebe-5bb4-41ab-a0f6-18bbab0ca4b4"}
	
	m = MultipartEncoder(fields, boundary='&')
	r = requests.post(url, headers={'Content-Type': m.content_type}, data=m.to_string())
	
	temp = json.loads(r.text)
	return temp["answer"]
	
print(send())
```
